framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -736,8 +736,11 @@ protected void addInformationFromPreconditions(
                         } else {
                             thenStore.insertValue(secondInternal, firstValue);
                         }
-                        return new ConditionalTransferResult<>(
-                                res.getResultValue(), thenStore, elseStore);
+                        // To handle `(a = b = c) == x`, repeat for all insertable receivers of
+                        // splitted assignments instead of returning.
```

Comment:
Yes. I already have a test case. `testStrings2` in `framework/tests/value/Switch.java` fails without this change.